src/System.Net.Http/tests/FunctionalTests/ResponseStreamTest.cs

```diff
@@ -68,6 +69,26 @@ public ResponseStreamTest(ITestOutputHelper output)
             }
         }
 
+        [Fact]
+        public async Task ReadAsStreamAsync_Cancel_TaskIsCanceled()
+        {
+            var cts = new CancellationTokenSource();
+
+            using (var client = new HttpClient())
+            {
+                HttpResponseMessage response =
+                    await client.GetAsync(HttpTestServers.RemoteEchoServer, HttpCompletionOption.ResponseHeadersRead);
+                Stream stream = await response.Content.ReadAsStreamAsync();
+
+                var buffer = new byte[2048];
+                Task task = stream.ReadAsync(buffer, 0, buffer.Length, cts.Token);
```

Comment:
I'll go with:
`((IAsyncResult)task).AsyncWaitHandle.WaitOne();` and pass in a TimeSpan for the `WaitOne`